### DISCUSSION TITLE
Git functionality

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#2070] [Q&A] [answered]
I never knew about the Git functionality. How does one use it? I searched in `gCtrl+H`, `gzCtrl+H`, I tried `-f git`, I tried `vd my_repo` and nothing worked.

**Comment by saulpw** [comment]
It's in a separate app now, `vgit`: https://github.com/saulpw/visidata/tree/develop/visidata/apps/vgit

**Reply by cool-RR** [reply]
How does a user invoke `vgit`? I thought that maybe I could just run `vgit` in the shell but that doesn't work.

**Reply by cool-RR** [reply]
Oh, should I install it like it says in the readme? `pip3 install git+git://github.com/visidata/vdplus/vgit@develop`

**Reply by saulpw** [reply]
Yeah, but I see now that's very old.  If you have a git clone of the visidata repo, `pip3 install -e visidata/apps/vgit`.

**Reply by cool-RR** [reply]
I installed it with your command now. It seems to be successfully installed, but it's not clear how to use it. The readme says "`vgit [<working_dir>]` automatically opens a git status sheet if the directory has a `.git` subdirectory." However, when I run `vgit` it seems to act as a pass-through to `git`:

```
➜  /herring vgit nichos
git: 'nichos' is not a git command. See 'git --help'.
➜
```

Am I missing some information source that shows how to use vgit?

Also, how do you intend real users to install vgit?

**Reply by saulpw** [reply, marked as answer]
I thought I had written up more about how to use it when I reworked it earlier this year while importing it into the main VisiData repo, but I can't find it.  Yes, it's like a git passthrough, which co-opts some subset of commands for a visual interface.  Try `vgit log` or `vgit branch` or `vgit status`.  Some of these interfaces are more useful than others.  I tried setting `alias git=vgit` for awhile myself but got annoyed with it.  Now I use it in limited circumstances when I want a data-oriented view of some git repo.

I think there might still be some value there, but it needs "more" time and attention (always more), and it's not worth it for me to invest more into it, other than keeping it running as an interesting example of the diversity of realms VisiData can provide an interface for.  I enjoy making things like this but there's only so many hours in a day and months in a lifetime, and vgit has quite a limited audience.

So if you're interested in playing with it and putting together a guide of sorts, I can give you a tour and point you in the right direction and hunt around for those docs I thought I wrote.  It'd be easier than it seems; I have put a few months of work into it, and it's actually reasonably put together.  But if you want it to be a polished product like VisiData, it's going to be disappointing, and you should just install tig or lazygit.

**Reply by cool-RR** [reply]
I see. I tried it now and it's pretty cool. (It didn't work for me before because I use `l` as an alias for `log` which `vgit` doesn't catch.) But yeah, like you I don't want to invest time in it.
